webapp/src/components/Analysis/PerturbationTestingPreview.test.tsx

```diff
@@ -0,0 +1,137 @@
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+
+describe("PerturbationTestingPreview", () => {
+  const handlers = [getPertubationResponse];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should have two toggle buttons for the train and test", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    await waitFor(() => {
+      expect(screen.getAllByRole("button")).toHaveLength(2);
+    });
+  });
+
+  it("should have a disabled toggle button if either one of the datasplit is unavailable", async () => {
+    renderPerturbationTestingPreview({ train: false, eval: true });
+    await waitFor(() => {
+      //Verifying if the train toggle button is disabled if not available in datasplits
+      expect(screen.getByRole("button", { pressed: false })).toHaveClass(
+        "Mui-disabled"
+      );
+      // By default, evaluation toggle is selected on page load.
+      expect(screen.getByRole("button", { pressed: true })).not.toHaveClass(
+        "Mui-disabled"
+      );
+    });
+  });
+
+  it("should display the data in the toggle with expected styling (color, font), format", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    await waitFor(() => {
+      // Verrify if the actual response data from API is visibly displayed on the toggle
+      expect(screen.getByText(/14.0%/i)).toBeVisible();
+      expect(screen.getByText("Failure rate - Evaluation Set")).toBeVisible();
+      expect(screen.getByText(/16.0%/i)).toBeVisible();
+      expect(screen.getByText("Failure rate - Training Set")).toBeVisible();
+
+      // styling should be applied for the selected toggle as expected
+      expect(screen.getByText(/14.0%/)).toHaveStyle(
+        "font-size: 3.75rem; color: rgb(237, 108, 2)"
+      );
+
+      // verify with some random unexpected values to not have been displayed.
+      expect(screen.queryByText(/17.0%/i)).toBeNull();
+      expect(screen.queryByText("some random text")).toBeNull();
+      expect(screen.queryByText(/26.0%/i)).toBeNull();
+    });
+  });
+
+  it("should not display a value if datasplit is not available", async () => {
+    renderPerturbationTestingPreview({ train: false, eval: true });
+    await waitFor(() => expect(screen.getByText("--%")).toBeVisible());
+  });
+
+  it("should have expected columns", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    const table_columns = [
+      "Test Family",
+      "Test Name",
+      "Modif. Type",
+      "Test Description",
+      "FR on Evaluation Set",
+    ];
+    const columnheaders = screen.getAllByRole("columnheader");
+    await waitFor(() => {
+      table_columns.forEach((name, index) => {
+        expect(columnheaders[index].textContent).toEqual(name);
+      });
+    });
```

Comment:
Sorry for [repeating](https://github.com/ServiceNow/azimuth/pull/215#discussion_r943693140), but `table_columns` should be camelCase. Also, I think `columnheaders` should be two words two. Anyway, I think the test would be clearer if we rename them to something like `expectedColumnHeaders` and `actualColumnHeaders`. Finally, I think the `waitFor` is necessary as the columns load even if there is not data, so
```suggestion
    const expectedColumnHeaders = [
      "Test Family",
      "Test Name",
      "Modif. Type",
      "Test Description",
      "FR on Evaluation Set",
    ];
    const actualColumnHeaders = screen.getAllByRole("columnheader");
    expectedColumnHeaders.forEach((name, index) => {
      expect(actualColumnHeaders[index].textContent).toEqual(name);
    });
```

Reply:
yes "waitFor" has been added as part of the upcoming change

Reply:
Sorry, I made a typo, I meant
> I think the `waitFor` is NOT necessary as the columns load even if there is not data

Hence my suggestion that was removing `waitFor`.